### PR TITLE
Clang format all the things

### PR DESCRIPTION
**Pull request by rictic**
So I tried out vscode to check whether it's typescript in general or atom in particular that's slow with the improved estree typings. Looks like it's atom in particular, as vscode doesn't lose a beat.

I also discovered that it's very easy to get it to run clang-format on save, so I decided I'd take a look to see how it would look if we adopted clang-format for the repo.

### CODE REVIEW

**Comment by FredKSchott** [comment]
personal opinion: Not familiar with clang-format, but I like the change overall. 1 thing I'm not a fan of: squashing all object literals into single lines. Is there a way to turn that off?

**Comment by rictic** [comment]
You can hint that you'd like it not to do that by ending a line with `//`. We may also be able to give it a style that we like better.

I generally like the changes, but not all of clang-format's opinions are my own. The thing that I like about an automatic formatter is it gives us a uniform style while saving us from having to think about it.

File: src/analyzer.ts [hunk 1]
```diff
@@ -81,7 +87,7 @@ export class Analyzer {
     if (this._documentDescriptors.has(url)) {
       return this._documentDescriptors.get(url);
     }
-    let promise = (async () => {
+    let promise = (async() => {
```
Comment (justinfagnani) [inline src/analyzer.ts]: this is unfortunate

File: src/analyzer.ts [hunk 2]
```diff
@@ -1,11 +1,15 @@
 /**
  * @license
  * Copyright (c) 2015 The Polymer Project Authors. All rights reserved.
- * This code may only be used under the BSD style license found at http://polymer.github.io/LICENSE.txt
- * The complete set of authors may be found at http://polymer.github.io/AUTHORS.txt
- * The complete set of contributors may be found at http://polymer.github.io/CONTRIBUTORS.txt
+ * This code may only be used under the BSD style license found at
```
Comment (justinfagnani) [inline src/analyzer.ts]: can we exclude @licence comments from formatting?

File: custom_typings/escodegen.d.ts [hunk 3]
```diff
@@ -3,13 +3,9 @@ declare module 'escodegen' {
   interface GenerateOpts {
     comment?: boolean;
     format?: {
-      indent?: {
-        style?: string;
-        base?: number;
-        adjustMultilineComment: boolean;
-      }
+      indent?:
+          {style?: string; base?: number; adjustMultilineComment: boolean;}
```
Comment (justinfagnani) [inline custom_typings/escodegen.d.ts]: are there any controls over literals besides the `//` hint?

File: src/analyzer.ts (hunk 2 above)
Comment (rictic) [inline src/analyzer.ts]: AFAICT, no: http://clang.llvm.org/docs/ClangFormatStyleOptions.html

File: src/analyzer.ts (hunk 1 above)
Comment (rictic) [inline src/analyzer.ts]: yep.

File: custom_typings/escodegen.d.ts (hunk 3 above)
Comment (rictic) [inline custom_typings/escodegen.d.ts]: Not afaict. I've reduced the wrapping some, but haven't found any tricks for object literals yet.

**Comment by rictic** [comment]
Got verbal LGTM from Justin. Merging

**Pull request merged**